Detectors/TRD/simulation/include/TRDSimulation/Digitizer.h

```diff
@@ -91,17 +100,25 @@ class Digitizer
 
   void getHitContainerPerDetector(const std::vector<HitType>&, std::array<std::vector<HitType>, kNdet>&);
   void clearCollections();
+  void setupSimulationValues();
 
   // Digitization chaing methods
   bool convertHits(const int, const std::vector<HitType>&, SignalContainer&, int thread = 0); // True if hit-to-signal conversion is successful
   bool convertSignalsToADC(const int, SignalContainer&, DigitContainer&, int thread = 0);     // True if signal-to-ADC conversion is successful
-
+  void addLabel(const o2::trd::HitType& hit, std::vector<o2::trd::MCLabel>&, std::unordered_map<int, int>&);
   bool diffusion(float, float, float, float, float, float, double&, double&, double&, int thread = 0); // True if diffusion is applied successfully
 
   // Helpers for signal handling
   static constexpr int KEY_MIN = 0;
   static constexpr int KEY_MAX = 2211727;
-  int calculateKey(const int det, const int row, const int col) { return ((det << 12) | (row << 8) | col); }
+  int calculateKey(const int det, const int row, const int col)
+  {
+    int key = ((det << 12) | (row << 8) | col);
+    if (key < KEY_MIN || key > KEY_MAX) {
+      LOG(FATAL) << "Wrong TRD key " << key << " for (det,row,col) = (" << det << ", " << row << ", " << col << ")";
```

Comment:
do you really need these checks? I think it might be better putting an assert statement that only fires in debug mode (or a unit test thoroughly the function). I looks to me that giving a wrong key should never happen and so the check at runtime is not useful and costly. It might for instance prohibit inlining.

Reply:
Yes, it is an unnecessary check, and we don't need it anymore. This should never happen by definition, but it will fail if someone eventually changes by mistake something elsewhere in the code. We put this there for testing. I will change it for an assert.